packages/amplify-graphql-model-transformer/src/__tests__/model-transformer.test.ts

```diff
@@ -1906,4 +1906,239 @@ describe('ModelTransformer:', () => {
+        const schema = `
+          type Foo @model {
+              id: Int
+              title: String!
+          }
+
+          type Bar @model {
+              id: Float
+              title: String!
+          }
+
+          type Baz @model {
+              id: Boolean
+              title: String!
+          }
+        `;
+
+        const out = testTransform({
+          schema,
+          transformers: [new ModelTransformer()],
+        });
+        expect(out.resolvers['Mutation.createFoo.init.1.req.vtl']).not.toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+        expect(out.resolvers['Mutation.createBar.init.1.req.vtl']).not.toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+        expect(out.resolvers['Mutation.createBaz.init.1.req.vtl']).not.toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+      });
+
+      it('should not include autoId when using a custom primary key', async () => {
+        const schema = `
+          type Post @model {
+              postId: ID! @primaryKey
+              title: String!
+          }
+        `;
+
+        const out = testTransform({
+          schema,
+          transformers: [new ModelTransformer(), new PrimaryKeyTransformer()],
+        });
+        expect(out.resolvers['Mutation.createPost.init.1.req.vtl']).not.toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+      });
+    });
+
+    describe('sql', () => {
+      it('should include autoId for basic ID', async () => {
+        const schema = `
+          type Post @model {
+              id: ID!
+              postId: ID! @primaryKey
+              title: String!
+          }
+        `;
+
+        const out = testTransform({
+          schema,
+          transformers: [new ModelTransformer(), new PrimaryKeyTransformer()],
+          dataSourceStrategies: constructDataSourceStrategies(schema, makeStrategy(MYSQL_DB_TYPE)),
+        });
+        expect(out.resolvers['Mutation.createPost.init.1.req.vtl']).toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+      });
+
+      it('should not include autoId for implicit ID', async () => {
+        const schema = `
+          type Post @model {
+              postId: ID! @primaryKey
+              title: String!
+          }
+        `;
+
+        const out = testTransform({
+          schema,
+          transformers: [new ModelTransformer(), new PrimaryKeyTransformer()],
+          dataSourceStrategies: constructDataSourceStrategies(schema, makeStrategy(MYSQL_DB_TYPE)),
+        });
+        expect(out.resolvers['Mutation.createPost.init.1.req.vtl']).not.toContain(
+          '$util.qr($ctx.stash.defaultValues.put("id", $util.autoId()))',
+        );
+      });
+
+      it('should include autoId when timestamps are null', async () => {
```

Comment:
Is it expected that we would generate an autoId for an explicitly declared `id` field even when there's an explicit primaryKey?

Reply:
That is the current behavior (as long as timestamps are not null) for dynamodb and sql. Changing this would be breaking now.